### ISSUE TITLE
Object constructor does not capture local values in annotations

### ISSUE THREAD

**Issue by rdhananjaya** [ballerina-platform/ballerina-lang#26838]
**Description:**
```ballerina
import ballerina/io;

type Foo record {|
    string s;
|};


annotation Foo foo on class;

string s = "g";

public function main() {
    s = "local";
    object {} o = @foo{ s } object { };

    var t = typeof o;
    io:println(t.@foo);
}
```

Output: `{"s":"g"}`

Expected `{"s":"local"}`

If we refer a local variable in the annotation, it gives the error `undefined symbol`.

**Steps to reproduce:**

**Affected Versions:**
SLP5

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by rdhananjaya** [comment]
We should be able to re-enable disabled test case on this issue. https://github.com/ballerina-platform/ballerina-lang/issues/26885

**Comment by rdhananjaya** [comment]
Fixed in  #27347
